smithy-model/src/test/java/software/amazon/smithy/model/selector/SelectorRunnerTest.java

```diff
@@ -0,0 +1,109 @@
+package software.amazon.smithy.model.selector;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.loader.Prelude;
+import software.amazon.smithy.model.node.ObjectNode;
+import software.amazon.smithy.model.shapes.Shape;
+import software.amazon.smithy.model.shapes.ShapeId;
+
+public final class SelectorRunnerTest {
+
+    @ParameterizedTest(name = "{0}")
+    @MethodSource("source")
+    public void selectorTest(Path filename) {
+        Model model = Model.assembler().addImport(filename).assemble().unwrap();
+        List<ObjectNode> tests = findTestCases(model);
+
+        for (ObjectNode test : tests) {
+            Selector selector = Selector.parse(test.expectStringMember("selector").getValue());
+            boolean skipPrelude = test.getBooleanMemberOrDefault("skipPreludeShapes", false);
+
+            Set<ShapeId> expectedMatches = new HashSet<>(new HashSet<>(test.expectArrayMember("matches")
```

Comment:
Why are there two hashset initializers here? I don't think you need either one of them. The outer one seems to just be making the set a HashSet, but you never append to it so it shouldn't be needed. I'm not sure what the inner one is for.